connector/src/main/scala/com/datastax/spark/connector/rdd/reader/PrefetchingResultSetIterator.scala

```diff
@@ -51,8 +53,11 @@ class PrefetchingResultSetIterator(resultSet: AsyncResultSet, timer: Option[Time
     currentIterator.hasNext || currentResultSet.hasMorePages
 
   override def next(): Row = {
-    maybePrefetch()
     val row = currentIterator.next()
+    maybePrefetch()
     row
   }
+
+  // See comment in maybePrefetch
+  maybePrefetch()
```

Comment:
I don't get why it fix anything, since you were calling maybePrefetch() in the next() method before the call to currentIterator.next().
Thus, even if the first page was empty, then the next having elements would have been prefetch.
Are you sure your UTs is failing without this change ?

Reply:
Yes, it is possible to have results made of pages with zero elements only